src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestShownTracker.ts

```diff
@@ -0,0 +1,108 @@
+
+import { Disposable, IDisposable } from '../../../../../base/common/lifecycle.js';
+import { IStorageService, StorageScope, StorageTarget } from '../../../../../platform/storage/common/storage.js';
+import { TerminalShellType } from '../../../../../platform/terminal/common/terminal.js';
+import { IExtensionService } from '../../../../services/extensions/common/extensions.js';
+
+
+export const TERMINAL_SUGGEST_DISCOVERABILITY_KEY = 'terminal.suggest.increasedDiscoverability';
+export const TERMINAL_SUGGEST_DISCOVERABILITY_COUNT_KEY = 'terminal.suggest.increasedDiscoverabilityCount';
+const TERMINAL_SUGGEST_DISCOVERABILITY_MAX_COUNT = 10;
+const TERMINAL_SUGGEST_DISCOVERABILITY_MIN_MS = 10000;
+
+interface ITerminalSuggestShownTracker extends IDisposable {
+	getFirstShown(shellType: TerminalShellType): { window: boolean; shell: boolean };
+	updateShown(): void;
+}
+
+export class TerminalSuggestShownTracker extends Disposable implements ITerminalSuggestShownTracker {
+	private _done: boolean;
+	private _count: number;
+	private _timeout: Timeout | undefined;
+	private _start: number | undefined;
+
+	private _firstShownTracker: { shell: Set<TerminalShellType>; window: boolean } | undefined = undefined;
+
+	constructor(
+		private readonly _shellType: TerminalShellType | undefined,
+		@IStorageService private readonly _storageService: IStorageService,
+		@IExtensionService private readonly _extensionService: IExtensionService
+
+	) {
+		super();
+		this._done = this._storageService.getBoolean(TERMINAL_SUGGEST_DISCOVERABILITY_KEY, StorageScope.APPLICATION, false);
+		this._count = this._storageService.getNumber(TERMINAL_SUGGEST_DISCOVERABILITY_COUNT_KEY, StorageScope.APPLICATION, 0);
+		this._register(this._extensionService.onWillStop(() => this._firstShownTracker = undefined));
+	}
+
+	get done(): boolean {
+		return this._done;
+	}
+
+	update(widgetElt: HTMLElement | undefined): void {
+		if (this._done) {
+			return;
+		}
+		this._count++;
+		this._storageService.store(TERMINAL_SUGGEST_DISCOVERABILITY_COUNT_KEY, this._count, StorageScope.APPLICATION, StorageTarget.MACHINE);
+		if (widgetElt && !widgetElt.classList.contains('increased-discoverability')) {
+			widgetElt.classList.add('increased-discoverability');
+		}
+		if (this._count >= TERMINAL_SUGGEST_DISCOVERABILITY_MAX_COUNT) {
+			this._setDone(widgetElt);
+		} else if (!this._start) {
+			this._start = Date.now();
+			this._timeout = setTimeout(() => {
+				this._setDone(widgetElt);
+			}, TERMINAL_SUGGEST_DISCOVERABILITY_MIN_MS);
+		}
+	}
+
+	private _setDone(widgetElt: HTMLElement | undefined) {
+		this._done = true;
+		this._storageService.store(TERMINAL_SUGGEST_DISCOVERABILITY_KEY, true, StorageScope.APPLICATION, StorageTarget.MACHINE);
+		if (widgetElt) {
+			widgetElt.classList.remove('increased-discoverability');
+		}
+		if (this._timeout) {
+			clearTimeout(this._timeout);
+			this._timeout = undefined;
+		}
```

Comment:
FYI `disposableTimeout` exists to make this nicer, don't need to bother with it though as this is throw away code.